application-tour-ui/src/main/resources/TourCode/Macros.xml

```diff
@@ -44,18 +44,19 @@
   <hidden>true</hidden>
   <content>{{velocity output='false'}}
 #*
- * Sort a list of XWiki objects using an property as comparator.
+ * Sort a list of XWiki objects using a property as comparator.
  *
  * @param inputObjects the list of XWiki objects to sort.
- * @param propertyName the name of the XWiki object property to sort the inputObjects by. Multiple values for the same property are allowed.
- * @param outputObjects the list where to append the sorted objects. This list must not be null and should be empty before calling the macro.
+ * @param propertyName the name of the XWiki object property to sort the inputObjects by. Multiple values for the same
+ * property are allowed.
+ * @param outputObjects the list where to append the sorted objects. This list must not be null and should be empty
+ * before calling the macro.
  *#
 #macro (sortXWikiObjectsList $inputObjects $propertyName $outputObjects)
   #set ($objectsMap = {})
   #foreach ($object in $inputObjects)
-    #set ($key = $object.getProperty($propertyName).value)
-    #set ($sameKeyObjects = $objectsMap.get($key))
```

Comment:
`$sameKeyObjects` is still being used below this change in
```
...
#set ($discard = $sameKeyObjects.add($object))
...
```

Even though it is being declared inside the `if` branch, it`s not a good practice to rely on the fact (feature/bug) that Velocity does not respect closures and everything is in a global scope.

This change should be reverted.